src/osd/scrubber/pg_scrubber.cc

```diff
@@ -949,10 +949,12 @@ void PgScrubber::select_range_n_notify()
 {
   get_counters_set().inc(scrbcnt_chunks_selected);
 
-  if (select_range()) {
+  if (select_range().has_value()) {
     // the next chunk to handle is not blocked
     dout(20) << __func__ << ": selection OK" << dendl;
-    m_osds->queue_scrub_chunk_free(m_pg, Scrub::scrub_prio_t::low_priority);
+    uint64_t num_chunk_objects = select_range().value();
```

Comment:
Doesn't this latest change result in calling select_range() twice?

Reply:
Sorry about that! Changed it now

Reply:
Considering that `num_chunk_objects` is not used outside the if..else block, you can consider making its scope tighter by using the if statement with initializer like I indicated originally. But in this case, since the method ends after the if..else block, I am okay if you leave it as is.